packages/storage/amplify_storage_s3_dart/lib/src/model/s3_item.dart

```diff
@@ -76,16 +75,32 @@ class S3Item extends StorageItem {
       key: key,
       lastModified: headObjectOutput.lastModified,
       eTag: headObjectOutput.eTag,
-      metadata: headObjectOutput.metadata?.toMap() ?? const {},
+      metadata: decodeMetadata(headObjectOutput.metadata?.toMap() ?? const {}),
       versionId: headObjectOutput.versionId,
       size: headObjectOutput.contentLength?.toInt(),
       contentType: headObjectOutput.contentType,
     );
   }
 
-  /// Removes `prefixToDrop` from `key` string.
-  ///
-  /// This should only be used internally.
+  /// Creates a [S3Item] from [s3.GetObjectOutput] provided by S3
+  /// Client.
+  @internal
+  factory S3Item.fromGetObjectOutput(
+    s3.GetObjectOutput getObjectOutput, {
+    required String key,
+  }) {
+    return S3Item(
+      key: key,
+      lastModified: getObjectOutput.lastModified,
+      eTag: getObjectOutput.eTag,
+      metadata: decodeMetadata(getObjectOutput.metadata?.toMap() ?? const {}),
```

Comment:
I would make this part of the `S3Item` constructor. Seems like it's easy to miss this but required on every instantiation.